File: WDF-Server/src/main/java/com/nohorang/suryongfootprint/model/Challenge.java
package com.nohorang.suryongfootprint.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Getter
@Setter
@Entity
@Table(name = "CHALLENGE")
public class Challenge {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int challengeId;

    @Column(name="title", nullable=false)
    private String title;

    @Column(name="info", nullable=false)
    private String info;

    @Column(name="condition", nullable=false)
    private int condition;

    @Column(name="participants", nullable=false)
    private int participants;
}

File: WDF-Android/app/src/main/java/kr/co/nohorang/suryongfootprint/data/Challenge.kt
package kr.co.nohorang.suryongfootprint.data

import com.google.gson.annotations.SerializedName
import java.io.Serializable

// JSON 타입 변환에 사용될 객체(DTO)
// request의 편의성을 고려해서 선언할 것.
data class Challenge (
                  @SerializedName("challengeId")var challenge_id: Int?,
                  @SerializedName("title")var title: String?,
                  @SerializedName("info")var info:String?,
                  @SerializedName("condition")var condition:Int?,
                  @SerializedName("participants")var participants:Int?
) :Serializable {
}
